packages/web3-common/src/json_rpc.ts

```diff
@@ -33,13 +34,22 @@ export const isResponseWithError = <Error = unknown, Result = unknown>(
 	(typeof response.id === 'number' || typeof response.id === 'string');
 
 export const isResponseWithNotification = <Result>(
-	response: JsonRpcNotification<Result>,
+	response: JsonRpcNotification<Result> | JsonRpcSubscriptionResult,
 ): response is JsonRpcNotification<Result> =>
 	!Array.isArray(response) &&
 	!!response &&
 	response.jsonrpc === '2.0' &&
 	response.params !== undefined;
 
+export const isSubscriptionResult = <Result>(
+	response: JsonRpcNotification<Result> | JsonRpcSubscriptionResult,
```

Comment:
we should only allow passing `JsonRpcSubscriptionResult` type here.

Reply:
If it's a single type, then there is no need of a type guard. Why we have this type guard is because the response we get is mix of these two types.